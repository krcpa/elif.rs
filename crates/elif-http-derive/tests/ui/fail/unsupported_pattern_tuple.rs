//! Test that tuple destructuring patterns are properly rejected

use elif_http_derive::{get, controller};

// Mock the required types
pub struct ElifRequest;
pub struct ElifResponse;
pub type HttpResult<T> = Result<T, Box<dyn std::error::Error>>;

#[controller("/api/test")]
pub struct TestController;

impl TestController {
    // This should fail because tuple destructuring is not supported
    #[get("/coords/{x}/{y}")]
    #[param(x: int)]
    #[param(y: int)]
    pub async fn get_coords(&self, (x, y): (u32, u32)) -> String {
        format!("Coords: {}, {}", x, y)
    }
}

fn main() {}