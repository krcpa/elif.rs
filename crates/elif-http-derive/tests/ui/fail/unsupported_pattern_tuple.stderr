error: Unsupported tuple destructuring pattern in parameter. Hint: Use individual parameters instead of tuple destructuring. Example: Change '(a, b): (Type1, Type2)' to 'a: Type1, b: Type2'.
  --> tests/ui/fail/unsupported_pattern_tuple.rs:18:9
   |
18 |     pub async fn get_coords(&self, (x, y): (u32, u32)) -> String {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
